main.go

```diff
@@ -22,6 +24,12 @@ func main() {
 	// parsing the flags has to be done after setting up all the flags
 	flag.Parse()
 
+	// If no flags are given, print the default usage message
+	if numFlags := flag.NFlag(); numFlags == 0 {
+		flag.PrintDefaults()
+		os.Exit(1)
+	}
```

Comment:
I think I understand what you were aiming for in here - making sure `file` was passed. However, it may be possible for `numFlags` to be 1 because the user sets the `prepper` flag, in which case we still do not have the `file` param; this causes it to not achieve its purpose.

Reply:
Another thing you can do is move the changes to `main.go` to a different branch to isolate the work. We can also discuss more details on that.

Reply:
> I think I understand what you were aiming for in here - making sure `file` was passed. However, it may be possible for `numFlags` to be 1 because the user sets the `prepper` flag, in which case we still do not have the `file` param; this causes it to not achieve its purpose.

Good catch!

> Another thing you can do is move the changes to `main.go` to a different branch to isolate the work. We can also discuss more details on that.

I think, I've addressed the issue: bef1871